rules.js

```diff
@@ -56,21 +59,12 @@ class StructuredData extends Rule {
     };
   }
 
-  setup() {
-    let start = -1;
-
-    this.on('tag:open', event => {
-      if (event.target.nodeName === 'script') {
-        start = event.target.location.offset;
-      }
-    });
-
-    this.on('tag:close', async event => {
-      const tag = event.previous;
-      if (event.target.nodeName === 'script' && tag.hasAttribute('type') && tag.getAttribute('type').value === 'application/ld+json') {
-        const startIdx = data.indexOf('>', start) + 1;
-        const endIdx = event.target.location.offset - 1; // omit the opening tag angled bracket
-        const content = data.substring(startIdx, endIdx);
+  
+setup() {
+    this.on('tag:close', event => {
+      const tag = event.target;
```

Comment:
For `tag:close` the event doesn't have a `target` property and that's why we need to remember the offset location of when the tag was opened. In the `tag:open` we also don't have access to attributes. This is why the logic here how it is.

Reply:
Thank you so much for finding the errors. I will look into it.